### ISSUE TITLE
Add a distributed cache implementation of IObjectCacheFactory

### ISSUE THREAD

**Issue by HeyJoel** [cofoundry-cms/cofoundry#46]
Currently we only have an in-memory implementation of `IObjectCacheFactory`. It should be fairly easy to add additional packages for solutions like redis.

**Comment by johnds1974** [comment]
Hi, has this feature been done? I ask because I am planning on using multiple Cofoundry instances, each on a vm in an Azure Virtual Machine scale set.

**Comment by HeyJoel** [comment]
No, not yet. If you're familiar with the Redis API it should be fairly trivial to create your own implementation of [`IObjectCacheFactory`](https://github.com/cofoundry-cms/cofoundry/blob/22507ed1946aaf279df59590b81a963d14b50dbc/src/Cofoundry.Core/Caching/IObjectCacheFactory.cs) and override the existing implementation using our [DI system](https://www.cofoundry.org/docs/framework/dependency-injection).

If you look at the [`InMemory implemetation`](https://github.com/cofoundry-cms/cofoundry/tree/master/src/Cofoundry.Core/Caching) you'll see it's a fairly bare bones abstraction.
